resources/windows/wix/customactions.wxs.erb

```diff
@@ -197,17 +197,56 @@ End If
 
     <!-- Due to PUP-6729, system may not have permission to modify DACL, so first take ownership -->
     <CustomAction
-        Id="SetTakeownPuppetPermissions"
-        Property="TakeownPuppetPermissions"
-        Value="&quot;[%WINDIR]\System32\cmd.exe&quot; /c IF EXIST &quot;[%ProgramData]\PuppetLabs\puppet&quot; &quot;[%WINDIR]\System32\takeown.exe&quot; /F &quot;[%ProgramData]\PuppetLabs\puppet\*&quot; /R /A /D N"
-        Execute="immediate"/>
-    <CustomAction
-        Id="TakeownPuppetPermissions"
-        BinaryKey="WixCA"
-        DllEntry="<%= @platform.architecture == "x64" ? "WixQuietExec64" : "WixQuietExec" %>"
-        Execute="deferred"
-        Return="check"
-        Impersonate="no"/>
+      Id="TakeownPuppetPermissions"
+      Script="vbscript"
+      Execute="deferred"
+      Return="check"
+      Impersonate="no">
+      <![CDATA[
+On Error Resume Next
+
+' Executes command, sending its stdout / stderr to the WScript host
+Sub ExecuteCommand (command)
+
+  WScript.StdOut.WriteLine("Executing Command : " & command)
+  Dim wshShell : Set wshShell = CreateObject("WScript.Shell")
+
+  ' Exec rather than Run to be able to capture stdout
+  Dim wshScriptExec : Set wshScriptExec = wshShell.Exec(command)
+
+  Do
+    WScript.StdOut.WriteLine(wshScriptExec.StdOut.ReadLine())
+  Loop While Not wshScriptExec.StdOut.AtEndOfStream
```

Comment:
Should this be `While not  AtEndOfStream`?  All of the examples I see prefer this style. e.g.

https://docs.microsoft.com/en-us/previous-versions/windows/internet-explorer/ie-developer/windows-scripting/skwz6sz4(v%3dvs.84)****

Reply:
The real question is... can `StdOut` stream be at the end before it's ever read? I suspect not and that an up-front check is not necessary here. That said, it's probably safer, and I don't see anything spelled out in the docs at https://docs.microsoft.com/en-us/previous-versions/windows/internet-explorer/ie-developer/windows-scripting/cbxxzwb5%28v%3dvs.84%29

Will update